controllers/cloud.redhat.com/providers/deployment/impl.go

```diff
@@ -270,6 +271,8 @@ func initDeployment(app *crd.ClowdApp, env *crd.ClowdEnvironment, d *apps.Deploy
 
 	d.Spec.Template.Spec.InitContainers = ics
 
+	d.Spec.Template.Spec.TerminationGracePeriodSeconds = pod.TerminationGracePeriodSeconds
```

Comment:
Why is this added in this PR? This doesn't look related to the lifecycle hooks at first glance.

Reply:
This is needed for hooks that can take a long time. We need this for Trino which will likely take about 10-20 minutes to fully shutdown because of the work it needs to do before it can say it is done with work. The default termination is 30 seconds, so it delete the pod before the pod was actually done.